src/Resource/Ssl/Ssl.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * @package Nexcess-SDK
+ * @license https://opensource.org/licenses/MIT
+ * @copyright 2018 Nexcess.net, LLC
+ */
+
+declare(strict_types  = 1);
+
+namespace Nexcess\Sdk\Resource\Ssl;
+
+use Nexcess\Sdk\ {
+  Resource\Client\Client,
+  Resource\Model
+};
+
+/**
+ * Certificates
+ */
+class Ssl extends Model {
+
+  /** {@inheritDoc} */
+  public const MODULE_NAME = 'Ssl';
+
+  /** {@inheritDoc} */
+  protected const _PROPERTY_ALIASES = [
+    'id' => 'cert_id'
+  ];
+
+  /** {@inheritDoc} */
+  protected const _PROPERTY_COLLAPSED = [];
+
+  /** {@inheritDoc} */
+  protected const _PROPERTY_MODELS = [
+    'client' => Client::class
+  ];
+
+  /** {@inheritDoc} */
+  protected const _PROPERTY_NAMES = [
+    'alt_names',
+    'approver_email',
+    'cert_id',
+    'chain',
+    'chain_crts',
+    'client_id',
+    'common_name',
+    'crt',
+    'csr',
+    'distinguished_name',
+    'domain',
+    'id',
+    'key',
+    'months',
+    'package_id'
+  ];
+
+  /** {@inheritDoc} */
+  protected const _READONLY_NAMES = [
+    'client_id',
+    'identity',
+    'is_expired',
+    'is_installable',
+    'is_multi_domain',
+    'is_real',
+    'is_wildcard',
+    'valid_from_date',
+    'valid_to_date'
+  ];
+
+  /**
+   * Create a new backup
+   * This method has two modes. If a CSR is present then it will attempt to
+   * create the cert via the CSR. Otherwise, it will attempt to create the csr
+   * and key and then create the certificate.
+   *
+   * @return Ssl
+   * @throws \GuzzleHttp\Exception\ClientException on fail
+   */
+  public function create() : Ssl {
+    $endpoint = $this->_getEndpoint();
+    assert($endpoint instanceof Endpoint);
+
+    if (!empty($this->get('csr'))) {
+      return $endpoint->createCertificateFromCsr(
+        $this->get('csr'),
+        $this->get('key'),
+        $this->get('months'),
+        $this->get('package_id'),
+        $this->get('approver_email')
+      );
+    }
+
+    return $endpoint->create(
+      $this->get('domain'),
+      $this->get('distinguished_name'),
+      $this->get('months'),
+      $this->get('package_id'),
+      $this->get('approver_email')
+    );
+  }
+
+  /**
+   * Import a certificate
+   *
+   * @return Ssl
+   * @throws \GuzzleHttp\Exception\ClientException on fail
+   */
+  public function import() : Ssl {
+    $endpoint = $this->_getEndpoint();
+    assert($endpoint instanceof Endpoint);
+
+      return $endpoint->importCertificate(
+        $this->get('key'),
+        $this->get('crt'),
+        $this->get('chain')
+      );
+  }
+
+  /**
+   * Decode an existing CSR and compare it to the package_id
+   *
+   * @return array
+   * @throws \GuzzleHttp\Exception\ClientException on fail
+   */
+  public function decodeCsr() : array {
```

Comment:
should this (and `getCsrDetails` below) be proxied?  seems to me like it would make sense for them to only exist on the endpoint.  if there's a reason they should be proxied, that's fine.

Reply:
I proxied them by default but yes, in hindsight, it only makes sense on the Endpoint. POSSIBLY a static method on Ssl?

Reply:
removed